client-tags/icon.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Icon client tag
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "James Wheare"
+    period: 2016
+    email: "james@irccloud.com"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `+icon` tag name. Instead, implementations SHOULD use the
+`+draft/icon` tag name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed tag name.
+
+## Introduction
+
+This specification defines a client-only message tag to indicate message icons.
+
+## Motivation
+
+To allow for richer display of messages with added visual context, a method for client and bot users to attach custom icons to messages is needed. This method is intended for message specific icons. For persistent avatars, a metadata key would be more appropriate.
+
+Visual clients are given a way to choose an appropriate icon resource for the display context, with a method for providing multiple icon sources, with different dimensions.
+
+A content type hint is also provided, to allow clients to filter unsupported image formats without downloading the resource.
+
+## Architecture
+
+### Dependencies
+
+Clients wishing to use this tag MUST negotiate the [`draft/message-tags`](../core/message-tags-3.3.html) capability with the server.
+
+### Format
+
+The icon tag is sent by a client with the client-only prefix `+` and its value is a JSON encoded array of image objects that reference icon resources for the message.
+
+    +draft/icon=<json>
+
+Image objects MAY contain any of the following members:
+
+| Member           | Description |
+| :--------------- | :---------- |
+| `src` (string)   | The Uniform Resource Indicator (URI) of an icon resource |
+| `sizes` (string) | A space separate list of pixel image dimensions in `wxh` format supported by the resource, or a value of `any` for scalable icons. Clients MAY use this to select an appropriate icon resource for the context and pixel density of the display |
+| `type` (string)  | A media type hint for the icon resource. Clients MAY use this to filter out unsupported icon types without downloading the resource |
```

Comment:
I think if a client has some strict requirements on the image format, they don't necessarily need to download the image. When the src is HTTP, the client can sent a HTTP request with an `Accept` header requesting the format(s) that the client supports. If the server cannot offer the src resource in any media types that the user asked for it SHOULD return an error.

Reply:
Note also, the server could offer the resource as multiple types depending on what the user asked for.